### ISSUE TITLE
0.13 garden validate does not catch missing `spec.image` or `build` field in a Deploy action

### ISSUE THREAD

**Issue by Walther** [garden-io/garden#3802]
Running `garden validate` in a project where the Deploy configuration does not specify either a `build` or `spec.image` field results in a validation pass.

Running `garden deploy` will result in a runtime error
```
Deploy type=container name=backend specifies neither a `build` nor `spec.image`
```

I have bumped into this repeatedly when converting examples from the 0.12 configuration syntax to 0.13 configuration syntax, so adding a tracking issue for this improvement.

**Comment by Walther** [comment]
Similarly, it is easy to leave the `dependencies` field in the wrong place when renaming `services` to `spec`. The `validate` command doesn't catch this, only the `deploy` command at runtime. (When converting the configs, the `dependencies` field needs to be lifted to the top level of `Deploy` action)

**Comment by vvagaytsev** [comment]
Fixed in #4021
